annotation-utils/src/main/java/edu/isi/vista/annotationutils/SerializedCasConverter.java

```
@@ -0,0 +1,198 @@
+
+      // mark spans as attached
+      attachedEventSpans.add(event);
+      attachedEventSpans.add(argument);
+
+      mapBuilder.put(event, new Argument(role, argument.getBegin(), argument.getEnd()));
+    }
+
+    // Create CTEvent list
+    ImmutableMap<CTEventSpan, Collection<Argument>> map = mapBuilder.build().asMap();
+    List<CTEvent> events = new ArrayList<>();
+    for (Map.Entry<CTEventSpan, Collection<Argument>> entry : map.entrySet()) {
+      // create and add event
+      CTEventSpan eventAnnotation = entry.getKey();
+      events.add(
+          new CTEvent(
+              eventType,
+              eventAnnotation.getBegin(),
+              eventAnnotation.getEnd(),
+              entry.getValue(),
+              eventAnnotation.getNegative_example()));
+    }
+
+    // catch events with no arguments
+    for (CTEventSpan event : JCasUtil.select(jcas, CTEventSpan.class)) {
+      if (!attachedEventSpans.contains(event)) {
+        events.add(
+            new CTEvent(
+                eventType,
+                event.getBegin(),
+                event.getEnd(),
+                Collections.emptyList(),
+                event.getNegative_example()));
+      }
+    }
+
+    // Using document ID to name the output files
+    // Temporary solution: Making the assumption document ID is present in the document text
+    String text = jcas.getDocumentText();
+    Pattern pattern = Pattern.compile("<DOC id=\"([^\"]*)\"");
+    Matcher matcher = pattern.matcher(text);
+    if (matcher.find()) {
+      String title = matcher.group(1);
+      String id = title
+              + "_"
+              + FilenameUtils.removeExtension(input.getName())
+              + ".json";
+      ObjectMapper mapper = new ObjectMapper();
+      try (OutputStream os = new FileOutputStream(new File(outputDir, id))) {
+        mapper.writeValue(os, events);
+      }
+    } else {
+      log.error("Missing document id: " + input.getAbsolutePath());
```

If this fails, just bubble to the top and crash. The worst possible thing which can happen in research code is silent errors.

---

This is gone because we are using title from document metadata.